# BF528 Final Project: ChIP-Seq snakemake workflow
Author: Johnathan Zhang

**View more detailed discussion in** `report/discussion_questions.html`

## Project Description
This project uses Chromatin Immunoprecipitation sequencing data targetting the RUNX1 transcription factor from [this article](https://pubmed.ncbi.nlm.nih.gov/27514584/).
The sequencing data is pushed through a Snakemake workflow for processing and exploratory analyses.
Results can be found in the report folder.

## Methods
Raw sequencing reads can be obtained via GEO Accession [GSE75070](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE75070).
Raw sequencing reads were first inspected for quality assessment using [FastQC v0.12.1](https://qubeshub.org/resources/fastqc).
The raw reads were then quality controlled with [Trimmomatic v0.39](http://www.usadellab.org/cms/?page=trimmomatic) using the arguments 'ILLUMINACLIP:TruSeq3-SE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15'. 
[Bowtie2 v2.5.3](https://bowtie-bio.sourceforge.net/bowtie2/index.shtml) `build` was used to build a genome index for human genome GRCh38
Post trimmomatic raw reads were then aligned to the generated reference using Bowtie2 as well.
Quality assessment for these alignments was performed using [Samtools v1.19.2](https://www.htslib.org/) `flagstat`to check for alignment quality. 
[MultiQC v1.20](https://multiqc.info/) was used to generate an overall quality report combining samtools flagstat and FastQC reports and can be found on the github repo https://github.com/BF528/project-2-John2018330.
Samtools was used to sorted and index the BAM's and then converted into bigwig files using [DeepTools  v](https://deeptools.readthedocs.io/en/develop/#) `bamCoverage`. 
A clustered Heatmap for the four samples was generated by DeepTools `multiBigWigSummary` and then `plotCorrelation` using `-c 'pearson' --plotNumbers --skipZeros`. The heatmap can be found below in section (...).
[HOMER v4.11](http://homer.ucsd.edu/homer/) was the primary software used for peak calling.
For each repliate of RUNX1 immunoprecipitation, HOMER `findPeaks` was used in conjunction with the non-enrichment sample (INP) with flags `-style factor -o auto`.
HOMER `pos2bed.pl` was used to turn the outputted peaks file to bed format.
[Bedtools v2.31.1](https://bedtools.readthedocs.io/en/latest/) `intersect` was first used with default flags to generate a list of reproducible peaks across the two RUNX1 replicates.
It was then used in conjunction with a file containing blacklisted regions with flag `-v` to remove known signal-artifact regions from the list of reproducible peaks.
HOMER `annotatePeaks.pl` was used to annotate peaks by providing both the GRCh38 primary assembly human genome and corresponding annotation gtf file.
HOMER `findMotifsGenome.pl` was used with default parameters to perform motif enrichment analysis with default parameters.
DeepTools `compute_matrix` was used to generate matrices of scores of peaks that fall into +/- 2kb regions  of interest defined by the HG38 annotation file using flags `-b 2000 -a 2000`.
DeepTools `plotProfile` was used to generate visualizations for both RUNX1 replicates showing signal coverage averaged across the body of all genes.
We combined differentially expressed genes identified via RNA-Seq analyses using DESeq2 with annotated peaks generated by HOMER to create figure 2F.


## Deliverables
All deliverables found in `report/discussion_questions.html`
1) MultiQC report combining QC reports about raw sequencing reads and alignment statistics
2) Correlation plot Heatmap showing overall similarity between sample files
3) Motif analysis conducted by Homer
4) Signal coverage plots of RUNX1 binding sites
5) Genome browser snapshot of the MALAT1 region and peak calling surrounding the region
